Samples/ResourceManagement/GraphRbac/ManageServicePrincipal/Program.cs

```diff
@@ -0,0 +1,194 @@
+using Microsoft.Azure.Management.Graph.RBAC.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using Microsoft.Azure.Management.Samples.Common;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace ManageServicePrincipal
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            try
+            {
+                //=================================================================
+                // Authenticate
+                AzureCredentials credentials = SdkContext.AzureCredentialsFactory.FromFile(Environment.GetEnvironmentVariable("AZURE_AUTH_LOCATION"));
+
+                var authenticated = Azure
+                    .Configure()
+                    .WithLogLevel(HttpLoggingDelegatingHandler.Level.Basic)
+                    .Authenticate(credentials);
+
+                RunSample(authenticated);
+            }
+            catch (Exception ex)
+            {
+                Utilities.Log(ex);
+            }
+        }
+
+        /**
+         * Azure Service Principal sample for managing Service Principal -
+         *  - Create an Active Directory application
+         *  - Create a Service Principal for the application and assign a role
+         *  - Export the Service Principal to an authentication file
+         *  - Use the file to list subcription virtual machines
+         *  - Update the application 
+         *  - Delete the application and Service Principal.
+         */
+        public static void RunSample(Azure.IAuthenticated authenticated)
+        {
+            string subscriptionId = authenticated.Subscriptions.List().First<ISubscription>().SubscriptionId;
+            Utilities.Log("Selected subscription: " + subscriptionId);
+            IActiveDirectoryApplication activeDirectoryApplication = null;
+            string authFilePath = Path.Combine(Utilities.ProjectPath, "Asset", "mySamplAuthdFile.azureauth").ToString();
+            try
+            {
+                activeDirectoryApplication = createActiveDirectoryApplication(authenticated);
+                createServicePrincipalWithRoleForApplicationAndExportToFile(authenticated, activeDirectoryApplication, BuiltInRole.Contributor, subscriptionId, authFilePath);
+                useAuthFile(authFilePath);
+                updateApplication(authenticated, activeDirectoryApplication);
+            }
+            finally
+            {
+                if (activeDirectoryApplication != null)
+                {
+                    // this will delete Service Principal as well
+                    authenticated.ActiveDirectoryApplications.DeleteById(activeDirectoryApplication.Id);
+                }
+            }
+        }
+
+        private static IActiveDirectoryApplication createActiveDirectoryApplication(Azure.IAuthenticated authenticated)
```

Comment:
Please fix first upper casing in all methods in this file

---
In reply to: [120696937](https://github.com/Azure/azure-sdk-for-net/pull/3329#discussion_r120696937) [](ancestors = 120696937)